src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,169 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final int INVALID_INDEX = -1;
+    private Node<T> head;
+    private Node<T> tail;
+    private int size = 0;
+
+    private static class Node<E> {
+        private E item;
+        private Node<E> previous;
+        private Node<E> next;
+
+        public Node(Node<E> previous, E item, Node<E> next) {
+            this.item = item;
+            this.previous = previous;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        linkLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("This index '" + index + "' are wrong!");
+        }
+        if (index == size) {
+            linkLast(value);
+        } else {
+            linkBefore(value, takeNode(index));
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T item: list) {
+            add(item);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return takeNode(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        T oldElement = takeNode(index).item;
+        takeNode(index).item = value;
+        return oldElement;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        T oldElement = get(index);
+        if (index == 0) {
+            unlinkHead();
+        } else {
+            unlinkAfterHead(index);
+        }
+        return oldElement;
     }
 
     @Override
     public boolean remove(T object) {
+        int objectIndex = takeNodeIndexByObject(object);
+        if (objectIndex == 0) {
+            unlinkHead();
+            return true;
+        }
+        if (objectIndex > 0) {
+            unlinkAfterHead(objectIndex);
+            return true;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private Node<T> takeNode(int index) {
+        Node<T> node = head;
+        for (int i = 0; i < index; i++) {
+            node = node.next;
+        }
+        return node;
+    }
+
+    private int takeNodeIndexByObject(T object) {
+        for (int i = 0; i < size; i++) {
+            if (takeNode(i).item == null && takeNode(i).item == object
```

Comment:
Don't create repeating code. You are calling `takeNode(i)` multiple times within the same iteration. Store the result in a variable and use that variable instead.